classes/controllers/grid/users/reviewer/PKPReviewerGridHandler.inc.php

```diff
@@ -963,30 +963,32 @@ public function gossip($args, $request)
      * @param array $args
      * @param PKPRequest $request
      *
-     * @return JSONMessage JSON object
+     * @return void|JSONMessage JSON object
      */
     public function fetchTemplateBody($args, $request)
     {
-        $template = new SubmissionMailTemplate($this->getSubmission(), $request->getUserVar('template'));
+        $context = $request->getContext();
+        $mailable = new class([$context, $this->getSubmission()]) extends Mailable
+        {
+            use Sender;
+        };
+        $template = $mailable->getTemplate($context->getId(), $request->getUserVar('template'));
+
         if (!$template) {
             return;
         }
 
         $user = $request->getUser();
-        $dispatcher = $request->getDispatcher();
-        $context = $request->getContext();
-
-        $template->assignParams([
-            'contextUrl' => $dispatcher->url($request, PKPApplication::ROUTE_PAGE, $context->getPath()),
-            'editorialContactSignature' => $user->getContactSignature(),
-            'signatureFullName' => $user->getFullname(),
-            'passwordResetUrl' => $dispatcher->url($request, PKPApplication::ROUTE_PAGE, $context->getPath(), 'login', 'lostPassword'),
+        $mailable->sender($user);
+        $mailable->setData(PKPLocale::getLocale());
+        $mailable->addData([
             'messageToReviewer' => __('reviewer.step1.requestBoilerplate'),
             'abstractTermIfEnabled' => ($this->getSubmission()->getLocalizedAbstract() == '' ? '' : __('common.abstract')), // Deprecated; for OJS 2.x templates
         ]);
-        $template->replaceParams();
 
-        return new JSONMessage(true, $template->getBody());
+        $body = Mail::compileParams($template->getLocalizedData('body'), $mailable->viewData);
```

Comment:
I think there are a few places where `->viewData` is used directly. There's now a `getData` method you can use:

```php
$mailable->getData($locale);
```

I think this should make it unnecessary to do `$mailable->setData($locale);`.